fe/src/main/java/org/apache/doris/alter/MaterializedViewHandler.java

```diff
@@ -1015,13 +1017,8 @@ private void runOldAlterJob() {
                 continue;
             }
 
-            db.writeLock();
-            try {
-                OlapTable olapTable = (OlapTable) db.getTable(rollupJob.getTableId());
-                rollupJob.cancel(olapTable, "cancelled");
-            } finally {
-                db.writeUnlock();
-            }
+            OlapTable olapTable = (OlapTable) db.getTable(rollupJob.getTableId());
```

Comment:
Why don't add the table lock？

Reply:
because olap table could be null, so I add write lock in the cancel function
  if (olapTable != null) {
            olapTable.writeLock();
            try {
                Preconditions.checkState(olapTable.getId() == tableId);
                for (Partition partition : olapTable.getPartitions()) {
                    if (partition.getState() == PartitionState.ROLLUP) {
                        partition.setState(PartitionState.NORMAL);
                    }
                }

                if (olapTable.getState() == OlapTableState.ROLLUP) {
                    olapTable.setState(OlapTableState.NORMAL);
                }
            } finally {
                olapTable.writeUnlock();
            }
        }